libs/libc/stdlib/lib_exit.c

```diff
@@ -22,13 +22,38 @@
  * Included Files
  ****************************************************************************/
 
+#include <nuttx/config.h>
+
+#include <nuttx/atexit.h>
+#include <nuttx/compiler.h>
+
 #include <stdlib.h>
 #include <unistd.h>
 
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Public Data
+ ****************************************************************************/
+
+extern FAR void *__dso_handle weak_data;
+FAR void *__dso_handle = &__dso_handle;
```

Comment:
I'm really not sure if this line is needed
```
FAR void *__dso_handle = &__dso_handle;
```
I think it does not hurt, but still.
OMO we need just to leave only
```
extern FAR void *__dso_handle[] weak_data;
```
and remove `FAR void *__dso_handle = &__dso_handle;`

Reply:
I believe we also have to define dsohandle somewhere, but I can try without

Reply:
Ok. Just figured it out.
We need to keep as it is now. So `extern FAR void *__dso_handle weak_data;` is a reference to a linker hidden symbol and `FAR void *__dso_handle = &__dso_handle;` creates a `__dso_handle` symbol if it is not provided by the linker.

Reply:
= &__dso_handle; is resolved by linker, so we need both extern decl and definition. I put it here because this is always compiled.